File: core-impl/src/commonMain/kotlin/com/example/myapplication/datasource/FirebaseStorageDatasourceImplementation.kt
package com.example.myapplication.datasource

import com.github.lamba92.firebasemultiplatform.storage.FirebaseStorage

class FirebaseStorageDatasourceImplementation(private val firebaseStorage: FirebaseStorage) :
    FirebaseStorageDatasource {

    override suspend fun uploadFile(data: ByteArray, path: String) =
        firebaseStorage.getReference(path).putBytes(data).await().let { true }

    override suspend fun getFileUrl(path: String) =
        firebaseStorage.getReference(path).downloadUrl()

}

File: web-client/src/app/core-impl/datasources/firebase-storage-datasource-implementation.ts
import {FirebaseStorageDatasource} from '../../data/datasources/firebase-storage-datasource';
import {Observable} from 'rxjs';
import {AngularFireStorage} from '@angular/fire/storage';
import {map} from 'rxjs/operators';
import {Injectable} from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class FirebaseStorageDatasourceImplementation implements FirebaseStorageDatasource {

  constructor(private firebaseStorage: AngularFireStorage) {
  }

  getFileUrl(path: string): Observable<string> {
    return this.firebaseStorage.ref(path).getDownloadURL().pipe(
      map((value) => {
        if (typeof value === 'string' || value instanceof String) {
          return value as string;
        } else {
          throw Error('js suck, long live to Kotlin!');
        }
      })
    );
  }

  uploadFile(data: Uint8Array, path: string): Observable<boolean> {
    return new Observable(subscriber => {
      this.firebaseStorage.ref(path).put(data).then((task) => {
        subscriber.next(true);
        subscriber.complete();
      });
    });
  }

}
